kingpin/actors/aws/cloudformation.py

```diff
@@ -230,6 +232,33 @@ def _discover_default_params(self, template_body):
         }
         return default_params
 
+    def _strip_hash(self, template_str: str, as_obj: bool = False):
+        """Strips the hash from the template.
+
+        Note: This will also strip the "Outputs" section if no other output exists. This might cause
+        issues when diffiing a template that contains an outputs section with no outputs.
+        """
+
+        template_obj = json.loads(template_str)
+
+        if len(KINGPIN_CFN_HASH_OUTPUT_KEY) > 0:
+            if "Outputs" not in template_obj.keys() or not isinstance(
```

Comment:
I'm confused by this check. Wouldn't this quit the function and return if someone had a simple template that just had no outputs? Wouldn't we _want to add the outputs section_ in that case?

Reply:
I think I answered these two comments in standup but LMK if these were about something different. There are 2 phases that kingpin will go through. A diff phase and a push phase. In the diff phase kingpin will collect info from the local repo about what that stack template should be and also collect info from the AWS CFN API to see what the live stack template is at the present moment. In order to remove complication, kingpin will silently strip out this managed hash value before the diff phase. However, in order to avoid the PR description issue, we will add a new hash value back to the template that we push in the changeset during the push phase. (... and in the next update we will again need to strip the hash for diff'ing, etc.)